src/main/scala/io/iohk/ethereum/blockchain/sync/regular/HeadersFetcher.scala

```diff
@@ -84,7 +82,7 @@ object HeadersFetcher {
     Behaviors.setup(context => new HeadersFetcher(peersClient, syncConfig, supervisor, context))
 
   sealed trait HeadersFetcherCommand
-  final case class FetchHeaders(blockNumber: BigInt, amount: BigInt) extends HeadersFetcherCommand
+  final case class FetchHeaders(block: Either[BigInt, ByteString], amount: BigInt) extends HeadersFetcherCommand
```

Comment:
wouldn't an explicit command be easier to reason about?

Reply:
the purpose of this command is unchanged regardless the parameter being Left or Right, having 2 commands would only add cumbersomeness imo. But I'm fine with whatever the team prefers. What do you think @LeonorLunatech @jvdp @AurelienRichez ?

Reply:
I have a slight preference for having two explicit commands `FetchHeadersByNumber` and `FetchHeadersByHash` instead of using Either to overload an existing command. It will lead to a few more lines, but I find it easier to understand. I note that we have other instances of this pattern in `GetBlockHeaders` case class.

Reply:
This `Either` might've been leaked from the json RPC. I think that's not the best reason to do it this way but I'm otherwise neutral on it.

Do we actually need both, though? Aren't we always going to fetch by hash?

Reply:
we need to support both, yes